exchangelib/folders.py

```diff
@@ -1006,14 +1006,12 @@ def get_default_folder(self, folder_cls):
         # folder was found, try as best we can to return the default folder of type 'folder_cls'
         if not folder_cls.DISTINGUISHED_FOLDER_ID:
             raise ValueError("'folder_cls' %s must have a DISTINGUISHED_FOLDER_ID value" % folder_cls)
-        # Use cached distinguished folder instance, but only if cache has already been prepped. This is an optimization
-        # for accessing e.g. 'account.contacts' without fetching all folders of the account.
-        if self._subfolders:
```

Comment:
Why was this removed?

Reply:
WELL. This broke test `test_ews_folder_hierarchy_watcher_sync`: previously, we were calling GetFolder on root, then asking for this `root_messages_folder_id` https://github.com/nylas/cloud-core/blob/master/nylas/sync/test/mailsync/backends/ews/integration/test_ews_folder_hierarchy_watcher.py#L45 which would then call `self._folders_map.values` in order to generate `self._folders` on the EWSAccount. When we hide that call behind this if check, we don't generate `self._folders` on the EWSAccount, and end up making individual API calls to get each `WellknownFolder` subfolder of root, instead of generating them en masse here. It's not necessarily a bad if statement, it's just different behaviour that breaks tests and might have implications elsewhere in the codebase

Reply:
Ah, I see. I don't think it would have weird implications in our code base, but I do think it's weird that `get_folder()` just directly invokes `_folder_map` while `get_default_folder()` only invokes it if it's already been invoked and cached. I think this makes more sense :+1: